engines/osxnative/sandbox.go

```diff
@@ -0,0 +1,147 @@
+package osxnative
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"io"
+	"mime"
+	"net/http"
+	"os"
+	"os/exec"
+)
+
+// white list of environment variables that must
+// be passed to the child process
+var environmentWhitelist = []string{
+	"PATH",
+	"HOME",
+	"USER",
+	"SHELL",
+	"TMPDIR",
+	"PWD",
+	"EDITOR",
+	"LANG",
+	"LOGNAME",
+	"TERM",
+	"TERM_PROGRAM",
+}
+
+type sandbox struct {
+	context     *runtime.TaskContext
+	taskPayload *payload
+	env         *[]string
+	aborted     bool
+}
+
+func newSandbox(context *runtime.TaskContext, taskPayload *payload, env *[]string) *sandbox {
+	return &sandbox{
+		context,
+		taskPayload,
+		env,
+		false,
+	}
+}
+
+func downloadLink(link string) (string, error) {
+	resp, err := http.Get(link)
+
+	if err != nil {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	contentDisposition := resp.Header.Get("Content-Disposition")
+	_, params, err := mime.ParseMediaType(contentDisposition)
+
+	if err != nil {
+		return "", err
+	}
+
+	filename := params["filename"]
+	file, err := os.Create(filename)
+
+	if err != nil {
+		return "", err
+	}
+
+	defer file.Close()
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		os.Remove(filename)
+		return "", err
+	}
+
+	return filename, nil
+}
+
+func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
+	if s.aborted {
+		return resultset{success: false}, engines.ErrSandboxAborted
+	}
+
+	if s.taskPayload.Link != "" {
+		filename, err := downloadLink(s.taskPayload.Link)
+
+		if err != nil {
+			s.context.LogError(err)
+			return nil, engines.ErrNonFatalInternalError
+		}
+
+		defer os.Remove(filename)
+
+		err = os.Chmod(filename, 0777)
+
+		if err != nil {
+			s.context.LogError(err)
+			return nil, engines.ErrNonFatalInternalError
+		}
+	}
+
+	env := make([]string, len(*s.env), len(*s.env)+len(environmentWhitelist))
+	copy(env, *s.env)
+
+	// Use the host environment plus custom environment variables
+	for _, e := range environmentWhitelist {
+		value := os.Getenv(e)
+		if value != "" {
+			env = append(env, e+"="+value)
+		}
+	}
+
+	cmd := exec.Command(s.taskPayload.Command[0], s.taskPayload.Command[1:]...)
+	cmd.Env = env
+
+	output, err := cmd.CombinedOutput()
```

Comment:
We should do something where stdout and stderr are streams to taskContext.Log() so that it's written to the task log as thre is data streaming in and the live log can later display it.  This appears to dump it once the command is done running.